### PR TITLE
Added rabbitmq-server package

### PR DESCRIPTION
**Pull request by brian7704**
Fulfills package request #10273 and seems to pass the criteria in the package request template

### CODE REVIEW

File: packages/rabbitmq-server/build.sh
Comment (TomJo2000) [inline packages/rabbitmq-server/build.sh:13]: `TERMUX_PKG_SERVICE_SCRIPT` splits off 2 fields from the array.
https://github.com/termux/termux-packages/blob/da5dac7f36383188e1c5b63a67ca1427c101763a/scripts/build/termux_step_install_service_scripts.sh#L1-L12

It may be preferable to add elements one by one to avoid confusion, especially with multi-line services.
```suggestion
TERMUX_PKG_SERVICE_SCRIPT=("rabbitmq-server")
TERMUX_PKG_SERVICE_SCRIPT+=(
"if [ -f \"$TERMUX_ANDROID_HOME/.config/rabbitmq/rabbitmq.conf\" ]; then 
	CONFIG=\"$TERMUX_ANDROID_HOME/.config/rabbitmq/rabbitmq.conf\"; else 
	CONFIG=\"$TERMUX_PREFIX/etc/rabbitmq/rabbitmq.conf\"; fi\n\
	exec rabbitmq-server \$CONFIG 2>&1"
)
```

File: packages/rabbitmq-server/build.sh
Comment (TomJo2000) [inline packages/rabbitmq-server/build.sh:21]: Use `:` (`true`) as a no-op.
```suggestion
	:
```

**Review by TomJo2000 — CHANGES_REQUESTED** [review]
Some procedural notes:
- You can use  `./scripts/lint-packages.sh packages/rabbitmq-server/build.sh`
to lint the build script locally.
- Please keep your commits squashed, I'll leave instructions for how to do that below in case you are not familiar with the procedure.

---

<sup>(This is a pre-written, saved reply.)</sup>
Please make sure to keep your commits squashed.
For adding to a single commit you can use `git commit --amend`.
Since you already have multiple commits on your branch though,
you'll need to squash those with `git rebase -i HEAD~<n>` first.
<sup>(Where <code>\<n\></code> is the number of commits you want to modify.
Please make sure to only modify ***your*** commits.)</sup>
<a href="https://www.baeldung.com/ops/git-squash-commits#1-squash-the-last-x-commits">https://www.baeldung.com/ops/git-squash-commits#1-squash-the-last-x-commits</a>

<h1><em></em></h1> 

Since squashing or amending commits changes the git history you will need to force push any such changes.
e.g. `git push --force`,
or preferably `git push --force-with-lease=branchname`
to make sure you aren't clobbering any refs you haven't fetched locally yet.

**Comment by brian7704** [comment]
I think I made the requested changes correctly but let me know if anything else needs to be fixed. Thanks!

File: packages/rabbitmq-server/build.sh
Comment (TomJo2000) [inline packages/rabbitmq-server/build.sh:12]: There's a couple things that should be removed as they are not applicable for termux.
- We don't have support for VMWare, so `vmware-rabbitmq` is not usable.
<sup>(Unless it does something other than what I'd expect based on the name, but there is no man page to check)</sup>
- Unpatched versions of RabbitMQ would not work on Termux, so `rabbitmq-upgrade` is not usable.
- `rabbitmq-service` is for running RabbitMQ as a Windows service, so we shouldn't ship the man page for it.
- Same for `rabbitmq-echopid`, also Windows specific.
```suggestion
TERMUX_PKG_PLATFORM_INDEPENDENT=true
TERMUX_PKG_RM_AFTER_INSTALL="
bin/rabbitmq-upgrade
bin/vmware-rabbitmq
lib/rabbitmq/lib/rabbitmq_server-${TERMUX_PKG_VERSION}/escript/rabbitmq-upgrade
lib/rabbitmq/lib/rabbitmq_server-${TERMUX_PKG_VERSION}/escript/vmware-rabbitmq
share/man/man8/rabbitmq-upgrade.8.gz
share/man/man8/rabbitmq-service.8.gz
share/man/man8/rabbitmq-echopid.8.gz
"
```

**Comment by brian7704** [comment]
I made those changes and squashed the commits again

File: packages/rabbitmq-server/build.sh
Comment (TomJo2000) [inline packages/rabbitmq-server/build.sh:40]: Is there any particular reason to create `enabled_plugins` as an empty file?

File: packages/rabbitmq-server/build.sh
Comment (TomJo2000) [inline packages/rabbitmq-server/build.sh:44]: This seems to work out fine, but it might be preferable to be explicit that we're putting these in `${TERMUX_PREFIX}/bin` since there is no `/sbin` in Termux.
```suggestion
	cp -r sbin/* "${TERMUX_PREFIX}/bin"
```

File: packages/rabbitmq-server/build.sh
Comment (TomJo2000) [inline packages/rabbitmq-server/build.sh:9]: Why are we `HOSTBUILD`-ing anyway?
You overwrite the hostbuild step with a no-op down below anyway, so we aren't actually doing a hostbuild.

Might be a misunderstanding of what the host build actually is.

**Review by TomJo2000 — APPROVED** [review]
That's all my blockers resolved, just have a couple questions I wanna clear up before merging.

File: packages/rabbitmq-server/build.sh
Comment (TomJo2000) [reply packages/rabbitmq-server/build.sh:40]: Right that would be because it strip out empty directories in the final packaging step.
Putting a placeholder file in there works fine, I just wasn't sure if the filename had any significance.

File: packages/rabbitmq-server/build.sh
Comment (TomJo2000) [reply packages/rabbitmq-server/build.sh:9]: Host builds can be a bit difficult to wrap your head around.
They're not usually required for most packages.

It's a build of the package for the platform ***hosting*** the build system, for example our package build container.
This is sometimes necessary to generate certain parts of packages that cannot be cross-compiled properly.

File: packages/rabbitmq-server/build.sh
Comment (TomJo2000) [inline packages/rabbitmq-server/build.sh:38]: I just want to check that this works as expected and then this PR is ready for merge.

Comment (TomJo2000) [reply packages/rabbitmq-server/build.sh:38]: Everything seems to be where it should be.
image

So, ready to merge.

**Comment by TomJo2000** [comment]
Thank you for your contribution :+1:

<h1><em></em></h1> 

Oh one more procedural thing I noticed during merge.
We somewhat follow conventional commits in the Termux repositories.
<sup>https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#commit-guidelines</sup>
I've manually fixed the commit message during merge,
but for future contributions please try to follow the commit message form:
```yml
<type>(<packaging-channel>/<package>): <action/description>
<Expanded explanation (recommended especially for any sort of packaging fix)>
```

**Comment by TomJo2000** [comment]
I added some additional information to the comment above in case it is helpful to anyone else coming across this PR in the future.

**Pull request merged**
